### ISSUE TITLE
Stereo example custom app

### ISSUE THREAD

**Issue by bvanmieg** [dicengine/dice#79]
Hi Dan,

It would be nice to have an example of a stereo correlation custom-app with a series of images (or at least cpp file). Current custom app example is only for 2D and for a single image (ref + def).

regards,
Bart

**Comment by dicengine** [comment]
This would be very helpful, also a bit more complicated. I’ll take a look to see if this would be workable.

Thanks,
Dan

On Apr 17, 2018, at 3:42 PM, bvanmieg <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi Dan,

It would be nice to have an example of a stereo correlation custom-app with a series of images (or at least cpp file). Current custom app example is only for 2D and for a single image (ref + def).

regards,
Bart

**Comment by bvanmieg** [comment]
Hi Dan,
is there any update on this topic?
Regards
Bart

**Comment by dicengine** [comment]
Hi Bart,

Yes and no. The main issue is that there are so many ways to do triangulation/calibration for stereo as well as some complexity in doing the cross correlation necessary in the initial step. It doesn’t lend itself immediately to a simple app that you could do in a few lines of code. The file DICe_Main.cpp has the algorithm used for stereo analysis and could be modified or replicated to make a library, but it would involve some tricky parts.

Is your main goal to try a new stereo DIC algorithm or to plug an existing stereo algorithm into a larger workflow? If you could give me some more details about what you want to accomplish there might be another path.

Thanks,
Dan



On Feb 21, 2019, at 1:38 AM, bvanmieg <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi Dan,
is there any update on this topic?
Regards
Bart

**Comment by bvanmieg** [comment]
Hi Dan,

Thanks again for the prompt feedback.

My intention to use the stereo correlation in a larger workflow.
Let me try to explain:
I have a machine which is equipped with a fixed stereo camera setup. I calibrated the cameras in that setup and have a dedicated camera calibration parameter file which is located in a dedicated folder on the machines computer. I also have a fixed ROI and a fixed subset location predefined.
My program would grab pictures during the machine's process and store the picture with predefined naming conventions in a predefined location.
The program would then correlated the pictures (always using the same predefined ROI and correlation settings) and write the results as txt files to a results directory.
As the last step, the program would post-process the data and provide a kind of measurement report to the user.

To summarize: my program would only need the stereo correlation part of DICe which will each time run with the same camera calibration file, correlation settings,... I do not need a user interface or whatsoever.
I'm convinced that all the features are in the DICe software but am not able to make a custom-app that is able to do what I want.
Any help would be more than welcome.

Regards,
Bart

**Comment by dicengine** [comment]
Hi Bart,

I’ll think about a way we could add  a feature to DICe that would enable this kind of workflow, but in the mean time this sounds like the perfect job for a Python script. It seems all you would have to do is write a script that modifies the input.xml file to replace the image folder location and run the DICe executable from the command line. Depending on the post-processing that you want to do the Python script could be roughly five to ten lines of code.

We’ve had similar needs in the past and we’ve used a Python script to do roughly the same thing.

-Dan


On Feb 21, 2019, at 1:50 PM, bvanmieg <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi Dan,

Thanks again for the prompt feedback.

My intention to use the stereo correlation in a larger workflow.
Let me try to explain:
I have a machine which is equipped with a fixed stereo camera setup. I calibrated the cameras in that setup and have a dedicated camera calibration parameter file which is located in a dedicated folder on the machines computer. I also have a fixed ROI and a fixed subset location predefined.
My program would grab pictures during the machine's process and store the picture with predefined naming conventions in a predefined location.
The program would then correlated the pictures (always using the same predefined ROI and correlation settings) and write the results as txt files to a results directory.
As the last step, the program would post-process the data and provide a kind of measurement report to the user.

To summarize: my program would only need the stereo correlation part of DICe which will each time run with the same camera calibration file, correlation settings,... I do not need a user interface or whatsoever.
I'm convinced that all the features are in the DICe software but am not able to make a custom-app that is able to do what I want.
Any help would be more than welcome.

Regards,
Bart
